ow_power_system/src/power_system_node.cpp

```diff
@@ -90,7 +39,16 @@ int main(int argc, char* argv[]) {
 
   ROS_INFO_STREAM_NAMED("power_system_node", "power_draw_csv_path is set to: " << csv_path);
 
-  auto power_csv = load_csv(csv_path);
+  // Read battery data from a file.
+  auto power_data = read_file(csv_path);
+
+  // Create a configuration from a file
+  string config_path = ros::package::getPath("ow_power_system")+"/config/example.cfg";
+  ConfigMap config(config_path);
+  // Contruct a new prognoser using the prognoser factory. The prognoser
+  // will automatically construct an appropriate model, observer and predictor
+  // based on the values specified in the config.
+  auto prognoser = PrognoserFactory::instance().Create("ModelBasedPrognoser", config);
 
   // Retrieve our publication rate expressed in Hz
   double power_update_rate = csv_path.find("onewatt") != string::npos ? 0.1 : 1;  // if onewatt default to 0.1 Hz otherwise 1 Hz
```

Comment:
The **power_update_rate** was based on the name of the three precomputed csv files. If these files are no longer going to be an option (which doesn't seem the case since the code here has certain expectation of the loaded csv file) then this line should be changed to something like this ("onewatt is not an option and the condition will always evaluate to 1): 
```suggestion
  double power_update_rate = 1;
```

Reply:
True. This will be changed.

Reply:
Resolved.